pennylane/tape/transforms/adjoint.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for the adjoint transform."""
+
+from functools import wraps
+from pennylane.tape.tapes import QuantumTape
+
+
+def adjoint(fn):
+    """Create a function that applies the adjoint of the provided operation or template.
+
+    This transform can be used to apply the adjoint of an arbitrary sequence of operations.
+
+    .. UsageDetails::

```

Comment:
@Thenerdstation the usage details here are great and super clear!

However I will be a bit nitpicky to make it consistent with our other docstrings:

* Can you swap `.. UsageDetails::` with `Args:` and `Returns:`? We typically have those first, like how NumPy does it (arguments first, usage example after).

* Also, it would be best to have a minimal example prior to the usage details. This way there is an example that is not hidden. Since Usage Details is collapsed by default, more advanced examples can go there. E.g.,
  
  ```rest
  Docstring summary.

  Explanation and links.

  Args:

  Returns:

  **Example**

  Minimal example showing basic usage.

  .. UsageDetails::

      More advanced examples and usage.
  ```

More details on the docstring conventions we follow are here: https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html#docstrings